### DISCUSSION TITLE
component examples for button styles

### DISCUSSION THREAD

**Discussion by houbly** [wmcadigital/wmn-design-system#526] [Ideas]
There are a couple of classes I've noticed in the buttons css that aren't currently listed in the design system, can we document them?:
Full width (block) button `wmnds-btn--block`
Button styled as a link `wmnds-btn--link`

**Comment by daylesalmon** [comment]
[Button as link is now in the design system](https://designsystem.wmnetwork.co.uk/components/buttons/#button-as-link) (wmnds-btn--link).

In terms of a button displayed as a block element - full width (wmnds-btn--block)... @jkmnch, @KaterinaKir do you want this to go in the design system? Will this be a common use in future, or are we going to stick to having buttons display to the width of the content inside of it?

If we want to include it, could you come up with the what, when etc. and we can get it added!

**Reply by jkmnch** [reply]
I don't think it's necessary to add separate components for buttons that are full-width, as we already explain on the Buttons page that full-width buttons only apply to pages on mobile:

> ### Button Size
> - The button width on desktop should be proportional with the text inside the button. Desktop buttons are aligned to the left edge of the content container.
> - Buttons are full-width on mobile. This is so users can easily tap on a button without stretching their thumb.


The only Desktop button which is full-width is the CTA button in the Question form component, but I've raised a separate issue to get this changed to text-width. #636

**Reply by daylesalmon** [reply]
In that case, I think i'll add the modifier class with an example in the design system so developers know that if you want to make a button full width there is already a class for it and don't have to reinvent the wheel.

And the copy at the top of the page will say when it's acceptable to do so (mobile).
